beater/route_config.go

```diff
@@ -72,16 +73,27 @@ var V1Routes = map[string]v1Route{
 
 var V2Routes = map[string]v2Route{
 	V2BackendURL: v2BackendRoute,
-	V2RumURL:     {rumRouteType},
+	V2RumURL:     v2RumRoute,
 }
 
-var v2BackendRoute = v2Route{
-	routeType{
-		v2backendHandler,
-		systemMetadataDecoder,
-		func(*Config) transform.Config { return transform.Config{} },
-	},
-}
+var (
+	v2BackendRoute = v2Route{
+		routeType{
+			v2backendHandler,
+			systemMetadataDecoder,
+			func(*Config) transform.Config { return transform.Config{} },
+		},
+		func(c *Config) rate.LimitHandler { return c.rateLimitHandler },
```

Comment:
Do we need this extra function when we have `v2backendHandler` and `v2rumHandler`? seems like it would be simpler to handle this under the individual handlers (`v2backendHandler` and `v2rumHandler`) and then i think we could get rid of `route.rateLimitHandler(beaterConfig)` [here](https://github.com/elastic/apm-server/pull/1367/files#diff-9829d1c9b6183c7ba00ca10afb6376b9R160)